src/mesh/mesh.cpp

```diff
@@ -811,12 +811,9 @@ void Mesh::OutputMeshStructure(const int ndim,
   std::cout << "Number of logical  refinement levels = " << current_level << std::endl;
 
   // compute/output number of blocks per level, and cost per level
-  auto nb_per_plevel = std::make_unique<int[]>(max_level);
-  auto cost_per_plevel = std::make_unique<int[]>(max_level);
-  for (int i = 0; i <= max_level; ++i) {
-    nb_per_plevel[i] = 0;
-    cost_per_plevel[i] = 0;
-  }
+  auto nb_per_plevel = std::vector<int>(max_level + 1);
+  auto cost_per_plevel = std::vector<int>(max_level + 1);
```

Comment:
Not strictly necessary, but if you want to be very clear and verbose, you can explicitly pass the value to initialize to the constructor. Also, `auto` is not needed anymore ;)

```suggestion
  std::vector<int> nb_per_plevel(max_level + 1, 0);
  std::vector<int> cost_per_plevel(max_level + 1, 0);
```

Reply:
I prefer this also.